src/components/form/Form.tsx

```diff
@@ -115,3 +119,23 @@ function useRedirectToStoredPage() {
     }
   }, [currentPageId, currentViewCacheKey, isValidPageId, location, navigateToPage]);
 }
+
+/**
+ * Sets the expanded width for the current page if it is defined in the currently viewed layout-page
+ */
+function useSetExpandedWidth() {
+  const currentPageId = useCurrentView();
+  const expandedPagesFromLayout = useExpandedWidthLayouts();
+  const expandedWidthFromSettings = usePageSettings().expandedWidth;
+  const { setExpandedWidth } = useUiConfigContext();
+
+  useEffect(() => {
+    let defaultExpandedWidth = false;
+    if (currentPageId && expandedPagesFromLayout[currentPageId] !== undefined) {
+      defaultExpandedWidth = !!expandedPagesFromLayout[currentPageId];
+    } else if (expandedWidthFromSettings !== undefined) {
+      defaultExpandedWidth = expandedWidthFromSettings;
+    }
+    setExpandedWidth(defaultExpandedWidth);
```

Comment:
I assume this will override user settings and shrink the page width every time you navigate on a page (unless the target page has this setting). Possibly a quite annoying if a user wanted to make the setting stick, but still, I guess it works.